Feedback on this PR when compiling on Windows
```
(base) D:\rlafage\workspace\mnist>cargo build --features download
   Compiling mnist v0.5.0 (D:\rlafage\workspace\mnist)
warning: unused import: `std::os::windows::fs::MetadataExt`
  --> src\download.rs:89:17
   |
89 |             use std::os::windows::fs::MetadataExt;
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: use of deprecated function `std::thread::sleep_ms`: replaced by `std::thread::sleep`
   --> src\download.rs:103:29
    |
103 |                     thread::sleep_ms(10);
    |                             ^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

error[E0599]: no method named `size` found for struct `Metadata` in the current scope
   --> src\download.rs:101:41
    |
101 |                     current_size = meta.size() as usize;
    |                                         ^^^^ method not found in `Metadata`

For more information about this error, try `rustc --explain E0599`.
warning: `mnist` (lib) generated 2 warnings
error: could not compile `mnist` due to previous error; 2 warnings emitted
```

If I remove the download feature, I get:
```
(base) D:\rlafage\workspace\mnist>cargo build
   Compiling mnist v0.5.0 (D:\rlafage\workspace\mnist)
warning: unused variable: `base_url`
   --> src\lib.rs:404:17
    |
404 |             let base_url = if self.use_fashion_data {
    |                 ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_base_url`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: panic message is not a string literal
   --> src\lib.rs:433:13
    |
433 | /             format!(
434 | |                 "Total data set length ({}) greater than maximum possible length ({}).",
435 | |                 total_length, available_length
436 | |             )
    | |_____________^
    |
    = note: `#[warn(non_fmt_panics)]` on by default
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
433 ~
434 |                 "Total data set length ({}) greater than maximum possible length ({}).",
435 |                 total_length, available_length
436 ~
    |

warning: panic message is not a string literal
   --> src\lib.rs:562:9
    |
562 | /         format!(
563 | |             "Expected magic number {} got {}.",
564 | |             LBL_MAGIC_NUMBER, magic_number
565 | |         )
    | |_________^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
562 ~
563 |             "Expected magic number {} got {}.",
564 |             LBL_MAGIC_NUMBER, magic_number
565 ~
    |

warning: panic message is not a string literal
   --> src\lib.rs:572:9
    |
572 | /         format!(
573 | |             "Expected data set length of {} got {}.",
574 | |             expected_length, length
575 | |         )
    | |_________^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
572 ~
573 |             "Expected data set length of {} got {}.",
574 |             expected_length, length
575 ~
    |

warning: panic message is not a string literal
   --> src\lib.rs:599:9
    |
599 | /         format!(
600 | |             "Expected magic number {} got {}.",
601 | |             IMG_MAGIC_NUMBER, magic_number
602 | |         )
    | |_________^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
599 ~
600 |             "Expected magic number {} got {}.",
601 |             IMG_MAGIC_NUMBER, magic_number
602 ~
    |

warning: panic message is not a string literal
   --> src\lib.rs:609:9
    |
609 | /         format!(
610 | |             "Expected data set length of {} got {}.",
611 | |             expected_length, length
612 | |         )
    | |_________^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
609 ~
610 |             "Expected data set length of {} got {}.",
611 |             expected_length, length
612 ~
    |

warning: panic message is not a string literal
   --> src\lib.rs:620:9
    |
620 |         format!("Expected rows length of {} got {}.", ROWS, rows)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
620 -         format!("Expected rows length of {} got {}.", ROWS, rows)
620 +         "Expected rows length of {} got {}.", ROWS, rows
    |

warning: panic message is not a string literal
   --> src\lib.rs:628:9
    |
628 |         format!("Expected cols length of {} got {}.", COLS, cols)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this usage of assert!() is deprecated; it will be a hard error in Rust 2021
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/panic-macro-consistency.html>
    = note: the assert!() macro supports formatting, so there's no need for the format!() macro here
help: remove the `format!(..)` macro call
    |
628 -         format!("Expected cols length of {} got {}.", COLS, cols)
628 +         "Expected cols length of {} got {}.", COLS, cols
    |

warning: unused return value of `Vec::<T, A>::split_off` that must be used
   --> src\lib.rs:460:9
    |
460 |         tst_img.split_off(tst_len * ROWS * COLS);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_must_use)]` on by default
    = note: use `.truncate()` if you don't need the other half

warning: unused return value of `Vec::<T, A>::split_off` that must be used
   --> src\lib.rs:461:9
    |
461 |         tst_lbl.split_off(tst_len);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: use `.truncate()` if you don't need the other half

warning: `mnist` (lib) generated 10 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.40s
```

--- a/src/download.rs
+++ b/src/download.rs
@@ -98,7 +98,12 @@ fn download(
                 while current_size < full_size {
                     let meta = fs::metadata(file_name.clone())
                         .expect(&format!("Couldn't get metadata on {:?}", file_name));
+                    
+                    #[cfg(target_family = "unix")]
                     current_size = meta.size() as usize;
+                    #[cfg(target_family = "windows")]
+                    current_size = meta.file_size() as usize;
+                    
                     pb.set(current_size.try_into().unwrap());
                     thread::sleep_ms(10);
                 }

--- a/src/download.rs
+++ b/src/download.rs
@@ -11,6 +11,21 @@ use pbr::ProgressBar;
 use std::convert::TryInto;
 use std::thread;
 
+#[cfg(target_family = "unix")]
+use std::os::unix::fs::MetadataExt;
+#[cfg(target_family = "windows")]
+use std::os::windows::fs::MetadataExt;
+
+#[cfg(target_family = "unix")]
+fn file_size(meta: &MetadataExt) -> usize {
+    meta.size() as usize
+}
+
+#[cfg(target_family = "windows")]
+fn file_size(meta: &MetadataExt) -> usize {
+    meta.file_size() as usize
+}
+
 const ARCHIVE_TRAIN_IMAGES: &str = "train-images-idx3-ubyte.gz";
 const ARCHIVE_TRAIN_IMAGES_SIZE: usize = 9912422;
 const ARCHIVE_TRAIN_LABELS: &str = "train-labels-idx1-ubyte.gz";
@@ -82,14 +97,7 @@ fn download(
             );
 
             let mut file = File::create(file_name.clone()).unwrap();
-
-            #[cfg(target_family = "unix")]
-            use std::os::unix::fs::MetadataExt;
-            #[cfg(target_family = "windows")]
-            use std::os::windows::fs::MetadataExt;
-
             let full_size = ARCHIVE_DOWNLOAD_SIZES[i];
-
             let pb_thread = thread::spawn(move || {
                 let mut pb = ProgressBar::new(full_size.try_into().unwrap());
                 pb.format("╢=> ╟");
@@ -98,12 +106,9 @@ fn download(
                 while current_size < full_size {
                     let meta = fs::metadata(file_name.clone())
                         .expect(&format!("Couldn't get metadata on {:?}", file_name));
-                    
-                    #[cfg(target_family = "unix")]
-                    current_size = meta.size() as usize;
-                    #[cfg(target_family = "windows")]
-                    current_size = meta.file_size() as usize;
-                    
+
+                    current_size = file_size(&meta);
+
                     pb.set(current_size.try_into().unwrap());
                     thread::sleep_ms(10);
                 }

